sigal/gallery.py

```diff
@@ -479,7 +489,11 @@ def zip(self):
         zip_gallery = self.settings['zip_gallery']
 
         if zip_gallery and len(self) > 0:
+            zip_gallery = zip_gallery.format(album=self)
             archive_path = join(self.dst_path, zip_gallery)
+            if isfile(archive_path):
+                self.logger.debug("Archive %s already created, passing", archive_path)
+                return zip_gallery
```

Comment:
The problem is exactly the same than copy big image. `{% if album.zip %}...{{ album.zip }}` in the template will call the generation 2 times.

The problem here is that we don't have an easy way to detect unchanged zip without generating the zip, which is very time consuming on my collection...